presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```diff
@@ -82,10 +83,10 @@ public PrometheusClient(PrometheusConnectorConfig config, JsonCodec<Map<String,
 
         tableSupplier = Suppliers.memoizeWithExpiration(metricsSupplier(metricCodec, getPrometheusMetricsURI(config)),
                 timeUnitsToSeconds(config.getCacheDuration()), TimeUnit.SECONDS);
+        MapType varcharMapType = (MapType) typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()));
```

Comment:
yes, this is much more compiler checkable, can see why they targeted this for fixing.